File: java/com/example/animalstepper/data/Conversion.java
package com.example.animalstepper.data;
public class Conversion {
    //inches to cm
    public static double inches_to_cm(double num) {
        return (double) (num * 2.54);
    }

    //cm to inches
    public static double cm_to_inches(double num) {
        return (double) (num / 2.54);
    }

    public static double ft_height_to_inches(double ft) {
        double num = (double) (12 * ft);
        return num;
    }

    public static double ft_in_height_to_inches(double ft, double inches) {
        double num = (double) (12 * ft) + inches;
        return num;
    }

    //height to stride base on unit and gender
    public static double height_to_stride(double height, String unit, char gender) {
        double inches;
        if (unit.equals("inches")) {
            inches = height;
        } else if (unit.equals("cm")) {
            inches = cm_to_inches(height);
        } else {
            System.out.println("The unit you enter is " + unit + ". Please enter either inches or cm");
            return 0;
        }

        double stride;
        if (gender == 'f') {
            stride = (double) (inches * 0.413);
        } else if (gender == 'm') {
            stride = (double) (inches * 0.415);
        } else {
            System.out.println("Invalid gender. Please enter either f or m");
            return 0;
        }

        return stride;
    }

    public static double meter_to_miles(double num) {
        return (double) (num / 1609.344);
    }

    public static double mile_to_meter(double num) {
        return (double) (num * 1609.344);
    }

    public static double km_to_miles(double num) {
        return (double) (num * 0.621371);
    }

    public static double miles_to_km(double num) {
        return (double) (num / 0.621371);
    }
}

File: old python files/conversion.py

#inches to  cm
def inches_to_cm(num):
    return float(num*2.54)

#cm to inches
def cm_to_inches(num):
        return float(num/2.54)

def ft_height_to_inches(ft):
    num = float(12*ft)
    return num

def ft_in_height_to_inches(ft,inches):
    num = float(12*ft)+inches
    return num

#height to stride base on unit and gender
def height_to_stride(height,unit,gender):
    if unit == "inches":
        #print(height)
        inches = height
    elif unit == "cm":
        inches = cm_to_inches(height)
    else:
        print("The unit you enter is " + str(unit) + " . Please enter either inches or cm")

    if gender == "f":
        stride = float(inches * 0.413)
    if gender == "m":
        stride = float(inches * 0.415)

    return stride
    
def meter_to_miles(num):
    return float(num/1609.344)

def mile_to_meter(num):
    return float(num*1609.344)

def km_to_miles(num):
    return float(num*0.621371)

def miles_to_km(num):
    return float(num/0.621371)